File: algorithms/Python/searching/linear_search.py
arr = [1, 4, 7, 9, 14, 17, 39, 56]
targets = (8, 39)


def linear_search(arr, target):
    """
    >>> all(linear_search(arr, x) == arr.index(x) if x in arr else -1 for x in targets)
    True
    """
    for i, item in enumerate(arr):
        if item == target:
            return i
    return -1


for target in targets:
    print(f"linear_search({arr}, {target}) = {linear_search(arr, target)}")

File: algorithms/JavaScript/src/searching/linear-search.js
// Linear Search JavaScript Version: searches for a
// specific item in an array and returns the index of the item.

function linearSearch(arr, x) {
  // Loop through the array
  for (let i = 0; i < arr.length; i++) {
    // Checks if the item in the array = search item.
    // If yes, return the index. If not, move to next item.
    if (arr[i] === x) {
      return i;
    }
  }
  // After looping if the search item is not found, return -1.
  return -1;
}

// Log to the console and have fun!
console.log(linearSearch([2, 5, 8], 5));
console.log(linearSearch([2, 6, 9, 15], 2));
console.log(linearSearch([2, 3, 4, 10], 8));
